src/gui/editorwidgets/qgsdatetimeeditwrapper.cpp

```diff
@@ -212,12 +212,19 @@ void QgsDateTimeEditWrapper::updateValues( const QVariant &value, const QVariant
     return;
 
   QDateTime dateTime;
+
   switch ( field().type() )
   {
     case QVariant::DateTime:
+      dateTime = value.toDateTime();
+      break;
     case QVariant::Date:
+      dateTime.setDate( value.toDate() );
+      dateTime.setTime( QTime::currentTime() );
```

Comment:
Not 00:00?

Reply:
It doesn't really matter, you can set it to whatever you want, it won't be seen or saved (or at least that's the idea).